### PR TITLE
API combined

### PR DESCRIPTION
**Pull request by malminawi**
MVP-prep branch merged with my current changes along with the admin page. please review. 
Thanks

### CODE REVIEW

File: api/models.py [hunk 1]
```diff
@@ -1,16 +1,42 @@
...
 from django.db import models
 from api.logistics.choice_enums import YEAR_IN_SCHOOL_CHOICES, LEARNING_GOALS_CHOICES, EXPERIENTIAL_LEARNING_HOURS_TYPES
 from api.reliability.validators import no_future_dates, hour_instance_validator, minutes_validator, student_id_validator
 
 from auth_backend.modules.common import models as common_models
+from auth_backend.modules.user.models import BaseVologUser
+
+
+class Mentor(models.Model):
+    #I added the foreign key as we have two users, student and mentor. Foreign key is used to distigush which user is which.
```
Comment (NullDefault) [inline api/models.py]: I don't think this is somewhere to use foreign keys. I met with Prof. Meng after cybersecurity to ask for advice regarding the two approaches (inheritance vs relations) and his stance was that it makes more sense to just do inheritance here.

File: api/models.py [hunk 2]
```diff
@@ -1,16 +1,42 @@
...
+    """
+    Represents a DAS mentor.
+    """
+    user = models.OneToOneField(BaseVologUser, on_delete=models.CASCADE)
+
+   # pass
+
+
+
+class Student(models.Model):
```
Comment (NullDefault) [inline api/models.py]: Both Student and Mentor should just subclass from the basevologuser class. I did it on my branch but I don't think you merged at that point

File: api/models.py [hunk 3]
```diff
@@ -1,16 +1,42 @@
...
+    """
+    Represents a Student user.
+    """
+    student_id = models.IntegerField(primary_key=True, unique=True, validators=[student_id_validator], blank=False)
+    class_standing = models.CharField(max_length=2, choices=[x.value for x in YEAR_IN_SCHOOL_CHOICES], blank=False)
+    DAS_mentor = models.ForeignKey(Mentor, on_delete=models.CASCADE, related_name="mentor",
+                                   blank=True, null=True)
+    user = models.OneToOneField(BaseVologUser, on_delete=models.CASCADE)
```
Comment (NullDefault) [inline api/models.py]: same as above. Look over how it's done on my branch.

File: auth_backend/modules/superAdmin/apps.py [hunk 4]
```diff
@@ -1,6 +1,5 @@
 from django.apps import AppConfig
 
-
-# all the admin stuff
+#all the admin stuff
 class AdminConfig(AppConfig):
-    name = 'modules.superAdmin'
+    name = 'apps.superAdmin'
```
Comment (NullDefault) [inline auth_backend/modules/superAdmin/apps.py]: why is this changed

File: auth_backend/modules/superAdmin/urls.py [hunk 5]
```diff
@@ -1,8 +1,14 @@
...
+from rest_framework.routers import DefaultRouter
+
+router = DefaultRouter()
+
+router.register('users-details', UserView, basename='user-details')
 
 urlpatterns = [
-    path('referrals/', CreateReferralView.as_view(), name='referral')
+    path('referrals/', CreateReferralView.as_view(), name='referral'),
+    path('', include(router.urls)),
```
Comment (NullDefault) [inline auth_backend/modules/superAdmin/urls.py:11]: what is this for

File: auth_backend/modules/user/apps.py [hunk 6]
```diff
@@ -2,4 +2,4 @@
 
 
 class UserConfig(AppConfig):
-    name = 'modules.user'
+    name = 'apps.user'
```
Comment (NullDefault) [inline auth_backend/modules/user/apps.py]: why was this changed to apps instead of modules

File: config/settings.py [hunk 7]
```diff
@@ -148,3 +148,16 @@
 STATICFILES_DIRS = (
     os.path.join(BASE_DIR, 'frontend', 'src', 'static'),
 )
+
+STATIC_ROOT = os.path.join(BASE_DIR, 'assets')
```
Comment (NullDefault) [inline config/settings.py]: this is  not where static root should be. All assets need to be inside of frontend/src/static/assets

File: config/settings.py [hunk 8]
```diff
@@ -148,3 +148,16 @@
...
+
+REST_FRAMEWORK = {
+    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.PageNumberPagination',
+    'PAGE_SIZE': 10
+}
+
+EMAIL_HOST = 'smtp.gmail.com'
+EMAIL_HOST_USER = 'mahmoud@alminawi.org'
```
Comment (NullDefault) [inline config/settings.py:158]: what is this and do we need a testing email as a placeholder

File: frontend/src/pages/Faculty/student/index.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+        searchQuery: ''
+    };
+    componentDidMount() {
+        this.getStudentData();
+    }
+
+
+
+
+    getStudentData = () => {
```
Comment (NullDefault) [inline frontend/src/pages/Faculty/student/index.js:21]: api queries should be in their own folder (logistics/services i believe).
Also the other queries are written differently (they use the fetch function)

**Review by NullDefault — CHANGES_REQUESTED** [review]

File: config/settings.py (hunk 8 above)
Comment (malminawi) [reply config/settings.py:158]: I did put the email as a placeholder to test. I did create a gmail testing account so my email won't be there.

File: auth_backend/modules/user/apps.py (hunk 6 above)
Comment (malminawi) [reply auth_backend/modules/user/apps.py]: it must be because of the merge. The modules folder was not renamed. Its fixed now.

File: auth_backend/modules/superAdmin/apps.py (hunk 4 above)
Comment (malminawi) [reply auth_backend/modules/superAdmin/apps.py]: it must be because of the merge. The modules folder was not renamed. Its fixed now.

File: api/models.py (hunk 1 above)
Comment (malminawi) [reply api/models.py]: done as requested
